src/routers/alpha-router/alpha-router.ts

```diff
@@ -119,51 +124,114 @@ export class AlphaRouter implements IRouter<AlphaRouterConfig>, ISwapToRatio<Alp
     token0Balance: CurrencyAmount,
     token1Balance: CurrencyAmount,
     position: Position,
+    swapAndAddConfig: SwapAndAddConfig,
     swapConfig?: SwapConfig,
     routingConfig = DEFAULT_CONFIG
   ): Promise<SwapRoute<TradeType.EXACT_INPUT> | null> {
-      if (
-        token0Balance.currency.wrapped.address.toLowerCase()
-        > token1Balance.currency.wrapped.address.toLowerCase()
-      ) {
+      if (token1Balance.currency.wrapped.sortsBefore(token0Balance.currency.wrapped)) {
         [token0Balance, token1Balance] = [token1Balance, token0Balance]
       }
 
-      const precision = JSBI.BigInt('1' + '0'.repeat(18))
-
-      const sqrtPriceX96 = position.pool.sqrtRatioX96
-      const sqrtPriceX96Lower = TickMath.getSqrtRatioAtTick(position.tickLower)
-      const sqrtPriceX96Upper = TickMath.getSqrtRatioAtTick(position.tickUpper)
-
-      const token0Proportion = SqrtPriceMath.getAmount0Delta(
-        sqrtPriceX96,
-        sqrtPriceX96Upper,
-        precision,
-        true
-      )
-      const token1Proportion = SqrtPriceMath.getAmount1Delta(
-        sqrtPriceX96,
-        sqrtPriceX96Lower,
-        precision,
-        true
-      )
+      // exit early for out of range positions
+      if (position.pool.tickCurrent < position.tickLower) {
```

Comment:
sorry a few questions here:

1. in this first case, where the current tick is below the lower, don't we want to be selling token1 for token0? and vice versa in the other case? as far as i can tell the code is reversed
2. assuming we sort out the above, i'm not sure we can early return. if we start out of range, i agree that we need to sell all of one asset for the other. but from working it out on paper, buying the asset we need will always push the market price closer to our desired range (assuming it executes in the pool we're targeting). so, simply selling all of our unwanted asset may push the price into our range, meaning our now 100/0 ratio is incorrect. am i missing something?

Reply:
1) i TOTALLY have this reversed. lol.
2) you're correct, bc I had this reversed, I also made this reverse assumption about where the price is moving.